From 9fbeb3229cc84e739cb111ed4b969d1af5c0e7bc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Fri, 3 Apr 2020 13:33:39 +0200
Subject: [PATCH 70/80] mainboard/pcengines: Use FMAP layout
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This commit removes the condition for the 'board.fmd' path, making it
the default path regardless of the VBOOT configuration. The changes have
been applied to the configurations of pcengines_apu2, pcengines_apu3,
pcengines_apu4, pcengines_apu5, and pcengines_apu6. A new file
'board.fmd' has also been added to 'src/mainboard/pcengines/apu2'.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
diff --git a/configs/config.pcengines_seabios_apu2 b/configs/config.pcengines_seabios_apu2
index fe3a9b3c2f6f..09bd18855197 100644
--- a/configs/config.pcengines_seabios_apu2
+++ b/configs/config.pcengines_seabios_apu2
@@ -20,6 +20,7 @@ CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,157b"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
+CONFIG_SEABIOS_BOOTORDER_IN_FMAP=y
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
 CONFIG_SEABIOS_REVISION_ID="8217fc68664a04ff4a86745ab0e9aa50aea06341"
diff --git a/configs/config.pcengines_seabios_apu3 b/configs/config.pcengines_seabios_apu3
index 8dc30fd9574c..0faaa445bb15 100644
--- a/configs/config.pcengines_seabios_apu3
+++ b/configs/config.pcengines_seabios_apu3
@@ -21,6 +21,7 @@ CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,1539"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
+CONFIG_SEABIOS_BOOTORDER_IN_FMAP=y
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
 CONFIG_SEABIOS_REVISION_ID="8217fc68664a04ff4a86745ab0e9aa50aea06341"
diff --git a/configs/config.pcengines_seabios_apu4 b/configs/config.pcengines_seabios_apu4
index 5960e6416835..7ac364122d3d 100644
--- a/configs/config.pcengines_seabios_apu4
+++ b/configs/config.pcengines_seabios_apu4
@@ -21,6 +21,7 @@ CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,1539"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
+CONFIG_SEABIOS_BOOTORDER_IN_FMAP=y
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
 CONFIG_SEABIOS_REVISION_ID="8217fc68664a04ff4a86745ab0e9aa50aea06341"
diff --git a/configs/config.pcengines_seabios_apu6 b/configs/config.pcengines_seabios_apu6
index 8796bebf4685..387b68b62d11 100644
--- a/configs/config.pcengines_seabios_apu6
+++ b/configs/config.pcengines_seabios_apu6
@@ -21,6 +21,7 @@ CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,1539"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
+CONFIG_SEABIOS_BOOTORDER_IN_FMAP=y
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
 CONFIG_SEABIOS_REVISION_ID="8217fc68664a04ff4a86745ab0e9aa50aea06341"
diff --git a/src/mainboard/pcengines/apu2/Kconfig b/src/mainboard/pcengines/apu2/Kconfig
index fbf29a452d8f..0561dcdfa90c 100644
--- a/src/mainboard/pcengines/apu2/Kconfig
+++ b/src/mainboard/pcengines/apu2/Kconfig
@@ -72,7 +72,7 @@ config APU2_PINMUX_OFF_C
 config APU2_PINMUX_GPIO0
 	bool "GPIO"
 	depends on BOARD_PCENGINES_APU2 || BOARD_PCENGINES_APU3 || \
-		BOARD_PCENGINES_APU4 || BOARD_PCENGINES_APU6 
+		BOARD_PCENGINES_APU4 || BOARD_PCENGINES_APU6
 
 config APU2_PINMUX_UART_C
 	bool "UART 0x3e8"
@@ -89,7 +89,7 @@ config APU2_PINMUX_OFF_D
 config APU2_PINMUX_GPIO1
 	bool "GPIO"
 	depends on BOARD_PCENGINES_APU2 || BOARD_PCENGINES_APU3 || \
-		BOARD_PCENGINES_APU4 || BOARD_PCENGINES_APU6 
+		BOARD_PCENGINES_APU4 || BOARD_PCENGINES_APU6
 
 config APU2_PINMUX_UART_D
 	bool "UART 0x2e8"
@@ -108,6 +108,7 @@ config FORCE_MPCIE2_CLK
 config FMDFILE
 	string
 	default "src/mainboard/\$(CONFIG_MAINBOARD_DIR)/vboot-rw-ab.fmd" if VBOOT
+	default "src/mainboard/\$(CONFIG_MAINBOARD_DIR)/board.fmd"
 
 config CBFS_SIZE
 	hex
diff --git a/src/mainboard/pcengines/apu2/board.fmd b/src/mainboard/pcengines/apu2/board.fmd
new file mode 100644
index 000000000000..0f344e580caa
--- /dev/null
+++ b/src/mainboard/pcengines/apu2/board.fmd
@@ -0,0 +1,18 @@
+FLASH 8M {
+	SI_BIOS@0x0 0x800000 {
+		BOOTORDER(PRESERVE)@0x0 0x1000
+		RW_VPD(PRESERVE)@0x1000 0x4000
+		SMMSTORE(PRESERVE)@0x5000 0x20000
+		RW_UNUSED@0x25000 0x1db000
+		WP_RO@0x200000 0x600000{
+			RO_VPD(PRESERVE)@0x0 0x4000
+			RO_SECTION@0x4000 0x5fc000{
+				FMAP@0x0 0x800
+				RO_FRID@0x800 0x40
+				RO_FRID_PAD@0x840 0x7c0
+				GBB@0x1000 0x40000
+				COREBOOT(CBFS)@0x41000 0x5bb000
+			}
+		}
+	}
+}
